packages/analytics-browser/template/browser-snippet.template.js

```diff
@@ -80,14 +74,37 @@ const snippet = (integrity, version) => `
       'setOptOut',
       'setTransport',
     ];
+    var funcsWithPromise = [
+      'init',
+      'add',
+      'remove',
+      'track',
+      'logEvent',
+      'identify',
+      'groupIdentify',
+      'setGroup',
+      'revenue',
+    ];
     function setUpProxy(instance) {
-      function proxyMain(fn) {
+      function proxyMain(fn, isPromise) {
         instance[fn] = function () {
-          instance._q.push([fn].concat(Array.prototype.slice.call(arguments, 0)));
+          var result = {
+            promise: new Promise((resolve) => {
+              instance._q.push({
+                name: fn,
+                args: Array.prototype.slice.call(arguments, 0),
+                resolve: resolve,
+              });
+            });
```

Comment:
```suggestion
            }),
```

Reply:
Good catch! My bad. I did not see it because they were same color in the vscode...